head/src/main/kotlin/io/qalipsis/core/head/report/ReportService.kt

```diff
@@ -60,5 +69,31 @@ internal interface ReportService {
      * @param size the maximum count of results in a page
      * @param page the 0-based index of the page to fetch
      */
-    suspend fun search(tenant: String, username: String, filters: Collection<String>, sort: String?, page: Int, size: Int): Page<Report>
+    suspend fun search(
+        tenant: String,
+        username: String,
+        filters: Collection<String>,
+        sort: String?,
+        page: Int,
+        size: Int
+    ): Page<Report>
+
+    /**
+     * Builds and renders a report in pdf form.
+     *
+     * @param tenant the reference of the tenant owning the report
+     * @param creator username of the currently authenticated user
+     * @param reference the unique identifier of the report
+     */
+    suspend fun render(tenant: String, creator: String, reference: String): String
```

Comment:
Do not return just a string, but the current state of the Report task, as object.